java/code/src/com/redhat/rhn/domain/action/Action.java

```diff
@@ -262,13 +264,28 @@ public void setServerActions(Set serverActionsIn) {
         this.serverActions = serverActionsIn;
     }
 
+    /**
+     * Retrieve the server action for the specified server, if exists
+     * @param serverId the id of the server
+     * @return the {@link ServerAction} for the specified server
+     */
+    public Optional<ServerAction> getServerActionForServer(Long serverId) {
```

Comment:
Seems it can have multiple actions for the requested server. You use "findFirst()". Wouldn't it be better to rename the function and call it "getFirstServerActionForServer" or return a List of all server actions.

Reply:
> Seems it can have multiple actions for the requested server

How is it possible? There is a unique constraint in `rhnserveraction` on (`server_id`, `action_id`)

```
rhn_server_action_sid_aid_uq" UNIQUE CONSTRAINT, btree (server_id, action_id)
```

Anyway, this method is a leftover of some testing I did, and it is not needed. I will remove it.

Reply:
Was this really limited to 1 specific action id ? If yes, than it is unique.
But anyway, not it is gone :-)